### ISSUE TITLE
geoip support

### ISSUE THREAD

**Issue by jonschipp** [jonschipp/mal-dnssearch#9]

**Comment by NexEng** [comment]
Could potentially use https://freegeoip.net/ API.

**Comment by jonschipp** [comment]
@NexEng Ah, yes that would be an easy fix.

**Comment by clinttepe** [comment]
The above API works, but is quite slow. So far using **geoiplookup** (geoip-bin dpkg) is the fastest way to achieve country-level lookups. Testing on a web server using real access log data resulted took less than a second per thousand IPs. The freegeoip.net API provided region-level data, but took nearly 100 times longer for the same IP list.

If the geoiplookup solution sounds acceptable, then I will finish out a PR.

**Comment by jonschipp** [comment]
@clinttepe Yes, geoiplookup is acceptable. We need to have it backwards compatible though so that if the necessary binary is not installed then it's not used.
